### ISSUE TITLE
Multiple PUT requests when updating same member lead to multiple members in GET response

### ISSUE THREAD

**Issue by fxjordan** [rwth-acis/RequirementsBazaar#122]
Sending a PUT multiple times with the same body to `https://beta.requirements-bazaar.org/bazaar/projects/95/members`:
```
[
	{
		"userId": 5823,
		"role": "ProjectMember"
	}
]
```

we get the same user multiple times in the member list:
```
[
  {
    "id": 5790,
    "userId": 5822,
    "role": "ProjectAdmin",
    "userProfileImage": "https://api.learning-layers.eu/profile.png",
    "userName": "netfelix"
  },
  {
    "id": 5794,
    "userId": 5823,
    "role": "ProjectMember",
    "userProfileImage": "https://api.learning-layers.eu/profile.png",
    "userName": "felix4debugging"
  },
  {
    "id": 5795,
    "userId": 5823,
    "role": "ProjectManager",
    "userProfileImage": "https://api.learning-layers.eu/profile.png",
    "userName": "felix4debugging"
  },
  {
    "id": 5796,
    "userId": 5823,
    "role": "ProjectMember",
    "userProfileImage": "https://api.learning-layers.eu/profile.png",
    "userName": "felix4debugging"
  },
  {
    "id": 5797,
    "userId": 5823,
    "role": "ProjectMember",
    "userProfileImage": "https://api.learning-layers.eu/profile.png",
    "userName": "felix4debugging"
  }
]
```

When we assign multiple roles to the same user this might be acceptable (even if, for example, ProjectManager inherits all privileges from ProjectMember). But we definitely should not add a user multiple times with the same role to the list of members.

**Comment by fxjordan** [comment]
Also, if we PUT to update an existing member, e.g. to downgrade the role from _ProjectManager_ to _ProjectMember_, a **new** member is added with the updated role, but the old member still exists. Thus, the user is still a _ProjectManager_ after the operation. Also, the same user is listed  multiple times in the list of members, which should be avoided.

After this API works in the right way, the first version of the member management UI is ready to be released

**Comment by fxjordan** [comment]
fixed in 9c408f12ccca5ad85be17b37744e9d92c8e6bda7

But we should refactor the `RoleRepository` code. Some method names are confusing and the concept of assigning some role in genera to a user is mixed up with the concept of adding/removing users to/from a project
